R/eval-tidy.R

```diff
@@ -342,3 +339,16 @@ f_self_eval <- function(overscope, overscope_top) {
     .Call(rlang_eval, f_rhs(f), overscope)
   }
 }
+tilde_eval <- function(f, env) {
+  if (is_env(attr(f, ".Environment"))) {
+    return(f)
+  }
+  f <- duplicate(f, shallow = TRUE)
+
+  # Inline the base primitive because overscopes override `~` to make
+  # quosures self-evaluate
+  mut_node_car(f, base::`~`)
```

Comment:
Maybe we need `new_language()`, which would have a bit more type checking than `node()`.